File: 1272-remove-interval/1272-remove-interval.java
class Solution {
    public List<List<Integer>> removeInterval(int[][] intervals, int[] rm) {
        List<List<Integer>> res = new ArrayList<>();
        
        for(int[] interval : intervals) {
           if(interval[1] < rm[0] || interval[0] > rm[1])
               res.add(Arrays.asList(interval[0], interval[1]));
            else {
                if(interval[0] < rm[0]) 
                    res.add(Arrays.asList(interval[0], rm[0]));
                if(interval[1] > rm[1]) 
                    res.add(Arrays.asList(rm[1], interval[1]));
            }
        }
        return res;
    }
}

File: 1272-remove-interval/1272-remove-interval.py
class Solution:
    def removeInterval(self, intervals: List[List[int]], toBeRemoved: List[int]) -> List[List[int]]:
        res = []
        intervals.sort(key = lambda x : x[0])
        for interval in intervals:
            if toBeRemoved:
                if interval[1] <= toBeRemoved[0]:
                    res.append(interval)
                elif interval[0] >= toBeRemoved[1]:
                    res.append(interval)
                    toBeRemoved = None
                else:#remove interval
                    if interval[0] < toBeRemoved[0]:
                        res.append([interval[0], toBeRemoved[0]])
                    if interval[1] > toBeRemoved[1]:
                        res.append([toBeRemoved[1], interval[1]])
                    
            else:
                res.append(interval)
        return res
        